buildpack/telemetry/fluentbit.py

```diff
@@ -0,0 +1,138 @@
+        util.upsert_custom_environment_variable(m2ee, var_name, value)
+
+
+def stage(buildpack_dir, destination_path, cache_path):
+
+    if not is_fluentbit_used():
+        return
+
+    logging.info("Fluentbit is being installed...")
+
+    util.resolve_dependency(
+        "fluentbit",
+        # destination_path - DOT_LOCAL_LOCATION
+        os.path.join(destination_path, NAMESPACE),
+        buildpack_dir=buildpack_dir,
+        cache_dir=cache_path,
+    )
+
+    for filename in (CONF_FILENAME, FILTER_FILENAME):
+        shutil.copy(
+            os.path.join(buildpack_dir, "etc", NAMESPACE, filename),
+            os.path.join(
+                destination_path,
+                NAMESPACE,
+            ),
+        )
+
+
+def update_config(m2ee):
+
+    if not is_fluentbit_used():
+        return
+
+    _set_default_env(m2ee)
+
+    util.upsert_logging_config(
+        m2ee,
+        {
+            "type": "tcpjsonlines",
+            "name": "FluentbitSubscriber",
+            "autosubscribe": "INFO",
+            "host": "localhost",
+            "port": str(LOGS_PORT),
+        },
+    )
+
+
+def run():
+
+    if not is_fluentbit_used():
+        return
+
+    fluentbit_dir = os.path.join(
+        os.path.abspath(".local"),
+        NAMESPACE,
+    )
+
+    fluentbit_bin_path = os.path.join(
+        fluentbit_dir,
+        "fluent-bit",
+    )
+
+    fluentbit_config_path = os.path.join(
+        fluentbit_dir,
+        CONF_FILENAME,
+    )
+
+    if not os.path.exists(fluentbit_bin_path):
+        logging.warning(
+            "Fluentbit is not installed yet. "
+            "Please redeploy your application to complete "
+            "Fluentbit installation."
+        )
+        return
+
+    logging.info("Starting Fluentbit...")
+
+    subprocess.Popen(
+        (fluentbit_bin_path, "-c", fluentbit_config_path),
+    )
+
+    # The runtime does not handle a non-open logs endpoint socket
+    # gracefully, so wait until it's up
+    @backoff.on_predicate(backoff.expo, lambda x: x > 0, max_time=10)
+    def _await_logging_endpoint():
+        return socket.socket(socket.AF_INET, socket.SOCK_STREAM).connect_ex(
+            ("localhost", LOGS_PORT)
+        )
+
+    logging.info("Awaiting Fluentbit log subscriber...")
+    if _await_logging_endpoint() == 0:
+        logging.info("Fluentbit log subscriber is ready")
+    else:
+        logging.error(
+            "Fluentbit log subscriber was not initialized correctly."
+            "Application logs will not be shipped to Fluentbit."
+        )
+
+
+def is_fluentbit_used():
+    """
+    The function checks if all environment variables
+    are set up which are required for Splunk connection.
+
+    """
+    required_env_vars = {
+        "SPLUNK_HOST",
+        "SPLUNK_TOKEN",
+    }
+
+    os_env_set = set(os.environ)
+
+    diff_envs = required_env_vars.difference(os_env_set)
+
+    if len(diff_envs) == 0:
+        return True
+    else:
+        return False
```

Comment:
Isn't it better to do the `os.getenv` check and return False if any missing ? Here we're fetching all the current environment vars unnecessarily.

A simplified version
```suggestion
return all(map(os.getenv, required_env_vars))
```

Reply:
It really looks much more simple, fixed